lib/recipe.rb

```diff
@@ -0,0 +1,26 @@
+class Recipe
+
+    attr_reader :name, :ingredients_required
+
+    def initialize(name)
+        @name = name
+        @ingredients_required = Hash.new(0)
+    end
+
+    def ingredients
+        @ingredients_required.keys
+    end
+
+    def add_ingredient(ingredient, qty)
+        @ingredients_required[ingredient] += qty
+    end
+
+    def total_calories
+        total_cal = 0
+        @ingredients_required.each do |ingredient|
+            cals = ingredient[0].calories * @ingredients_required[ingredient[0]]
```

Comment:
Using your two block variables, you already had the quantity, didn't need to go back into @ingredients_required here.  This does work though!